### ISSUE TITLE
SC2016 gives false positive in parameter substitution

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#501]
```shell
#!/bin/sh
VAR=foo
echo "${VAR:+TEXT='$VAR'}"
```
it outputs here correctly: TEXT='foo'
(hitted it in a here-document, where we write out hmtl-code, e.g.
```
#!/bin/sh
cat <<EOF
<body ${font_color:+ LINK='$font_color' VLINK='$font_color' ALINK='$font_color'}>
EOF
```

**Comment by koalaman** [comment]
Oh wow. The replacement is being parsed as if single quotes matter, but when it's embedded in a double quoted string they become literal. I would not have expected that.

**Comment by jasonkarns** [comment]
Is there a workaround here (besides disabling the rule for the line)?

**Comment by stdedos** [comment]
Dummy tc script:
```bash
#!/bin/bash

export var_0 var_1 var_2 var_3 var_4 var_5 
var_0="${out[0]:-'$var_0'}"
var_1="${out[1]:-'$var_1'}"
var_2="${out[2]:-'$var_2'}"
var_3="${out[3]:-'$var_3'}"
var_4="${out[4]:-'$var_4'}"
var_5="${out[5]:-'$var_5'}"
```

Should be clean
